### ISSUE TITLE
Sentry exception events occurring in main thread are not flushed to Sentry (fbs Pro feature)

### ISSUE THREAD

**Issue by apastel** [mherrmann/fbs#283]
Exceptions that are captured by Sentry in the main thread of execution are not being flushed to Sentry when my program exits. So, they do not show up in my Sentry project.

My hunch is that this is due to the `default_integrations` being set to `False` in `fbs_runtime/excepthook/sentry.py`. This disables Sentry's `Atexit` integration which is supposed to flush events from the background queue on interpreter shutdown. https://docs.sentry.io/platforms/python/configuration/integrations/default-integrations/

I've set up a minimal example adhering to the fman docs to demonstrate the behavior:
```
class MainWindow(QMainWindow, Ui_MainWindow):
    def __init__(self):
        super().__init__()

        1 / 0  # raises ZeroDivisionError

class AppContext(ApplicationContext):
    @cached_property
    def window(self):
        return MainWindow()

    @cached_property
    def exception_handlers(self):
        result = super().exception_handlers
        if is_frozen():
            result.append(self.sentry_exception_handler)
        return result

    @cached_property
    def sentry_exception_handler(self):
        return SentryExceptionHandler(
            PUBLIC_SETTINGS["sentry_dsn"],
            PUBLIC_SETTINGS["version"],
            PUBLIC_SETTINGS["environment"],
        )

    def run(self):
        self.window.show()
        return self.app.exec()

if __name__ == "__main__":
    appctxt = AppContext()
    appctxt.run()
```

 * Your operating system(s)
   * Windows 11
 * Your Python version
   * Python 3.10.6
 * Your fbs version
   * fbs_pro-1.1.3
 * Your PyInstaller version
   * 5.3
 * Your PyQt / PySide version
   * PySide 6.3.1

When I freeze and run the resulting .exe file, I am met with the expected exception, I close the window, and nothing shows up in my Sentry project (I've separately verified I can get exceptions to show up in my Sentry project by following their Python test example, or if the exception occurs in a Slot in my fbs program).
![image](https://user-images.githubusercontent.com/8836343/189420609-e0a03911-5662-47eb-adbf-bd6bfef6288f.png)

### What I've done to try to correct the problem
Inside `fbs_runtime/excepthook/sentry.py`, I firstly set `debug=True` in the `sentry_sdk.init()` call. When I run my program with the Sentry debug output, this is what I see in my application log:
```
 [sentry] DEBUG: Setting up integrations (with default = False)
Traceback (most recent call last):
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 41, in <module>
    appctxt.run()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 36, in run
    self.window.show()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 18, in window
    return MainWindow()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 13, in __init__
    1 / 0  # raises ZeroDivisionError
ZeroDivisionError: division by zero
 [sentry] DEBUG: Sending event, type:null level:error event_id:7bdbffc60485436287229d86afd66c08 project:6715501 host:o1393803.ingest.sentry.io
```
So there's a DEBUG message there at the end that says "Sending event", but nothing shows up in Sentry.  
Now I'm going to add the line `sentry_sdk.flush()` as the last line in the `handle()` function in `sentry.py` like so:
```
    def handle(self, exc_type, exc_value, enriched_tb):
        if self._rate_limiter.please():
            sentry_sdk.capture_exception((exc_type, exc_value, enriched_tb))
            sentry_sdk.flush()  # added by apastel
```
Here is the new program output:
```
 [sentry] DEBUG: Setting up integrations (with default = False)
Traceback (most recent call last):
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 41, in <module>
    appctxt.run()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 36, in run
    self.window.show()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 18, in window
    return MainWindow()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 13, in __init__
    1 / 0  # raises ZeroDivisionError
ZeroDivisionError: division by zero
 [sentry] DEBUG: Flushing HTTP transport
 [sentry] DEBUG: background worker got flush request
 [sentry] DEBUG: Sending event, type:null level:error event_id:b1a3d6c6b4ab46abac8e44080bde4067 project:6715501 host:o1393803.ingest.sentry.io
 [sentry] DEBUG: background worker flushed
```
Now there's several lines about flushing. And at the same time, the exception shows up right away in my Sentry project.
![image](https://user-images.githubusercontent.com/8836343/189421670-14b73647-ba8b-48ba-99fc-e31835553a9d.png)

So as I mentioned at the start, reading more about the Sentry [default integrations](https://docs.sentry.io/platforms/python/configuration/integrations/default-integrations/) shows me there's a number of integrations like `Atexit` and `Excepthook` that are being disabled by fbs, and I believe it's the `Atexit` one that normally does the flushing. I also tried just setting `default_integrations=True` instead of `False` without the explicit `flush()` call, and it looks like it tries to flush and then times out and fails for some unknown reason. 
```
 [sentry] DEBUG: Setting up integrations (with default = True)
...
 [sentry] DEBUG: Setting up previously not enabled integration logging
 [sentry] DEBUG: Setting up previously not enabled integration stdlib
 [sentry] DEBUG: Setting up previously not enabled integration excepthook
 [sentry] DEBUG: Setting up previously not enabled integration dedupe
 [sentry] DEBUG: Setting up previously not enabled integration atexit
 [sentry] DEBUG: Setting up previously not enabled integration modules
 [sentry] DEBUG: Setting up previously not enabled integration argv
 [sentry] DEBUG: Setting up previously not enabled integration threading
 [sentry] DEBUG: Setting up previously not enabled integration redis
 [sentry] DEBUG: Did not enable default integration redis: Redis client not installed
 [sentry] DEBUG: Enabling integration logging
 [sentry] DEBUG: Enabling integration stdlib
 [sentry] DEBUG: Enabling integration excepthook
 [sentry] DEBUG: Enabling integration dedupe
 [sentry] DEBUG: Enabling integration atexit
 [sentry] DEBUG: Enabling integration modules
 [sentry] DEBUG: Enabling integration argv
 [sentry] DEBUG: Enabling integration threading
 [sentry] DEBUG: Enabling integration redis
Traceback (most recent call last):
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 41, in <module>
    appctxt.run()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 36, in run
    self.window.show()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 18, in window
    return MainWindow()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\src\main\python\test_sentry.py", line 13, in __init__
    1 / 0  # raises ZeroDivisionError
ZeroDivisionError: division by zero
 [sentry] DEBUG: atexit: got shutdown signal
 [sentry] DEBUG: atexit: shutting down client
 [sentry] DEBUG: Flushing HTTP transport
Traceback (most recent call last):
  File "C:\Users\alexr\AppData\Local\Programs\Python\Python310\lib\threading.py", line 973, in _bootstrap
    self._bootstrap_inner()
 [sentry] DEBUG: background worker got flush request
  File "C:\Users\alexr\AppData\Local\Programs\Python\Python310\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\venv\lib\site-packages\sentry_sdk\integrations\threading.py", line 67, in run
    return old_run_func(self, *a, **kw)
  File "C:\Users\alexr\dev\ytmusic-deleter-packager\venv\lib\site-packages\fbs_runtime\excepthook\__init__.py", line 84, in run_with_except_hook
    run_original(*args2, **kwargs2)
TypeError: Thread.run() takes 1 positional argument but 2 were given
 [sentry] DEBUG: 4 event(s) pending on flush
Sentry is attempting to send 4 pending error messages
Waiting up to 2 seconds
Press Ctrl-Break to quit
 [sentry] ERROR: flush timed out, dropped 4 events
 [sentry] DEBUG: background worker flushed
 [sentry] DEBUG: Killing HTTP transport
 [sentry] DEBUG: background worker got kill request
 [sentry] DEBUG: Killing HTTP transport
 [sentry] DEBUG: background worker got kill request
```

So, my workaround for now is adding the `sentry_sdk.flush()` line after `sentry_sdk.capture_exception()`. Let me know if I'm doing anything wrong but I've read and re-read the fbs manual section on Sentry many times.

Thanks!

**Comment by apastel** [comment]
Hmm, yes actually. It looks like you can disable the default integrations and then re-enable individual ones:

```
from sentry_sdk.integrations.atexit import AtexitIntegration

def init(self):
        sentry_sdk.init(
            self._dsn, release=self._app_version, environment=self._environment,
            attach_stacktrace=True, default_integrations=False, integrations=[AtexitIntegration()]
        )
```

Tested this and it works. The ZeroDivisionError is flushed to Sentry after my app exits. 

I also tested registering an `atexit` handler in my application that calls `sentry_sdk.flush()`  and that worked as well.

**Comment by mherrmann** [comment]
Fixed in [fbs Pro 1.1.4](https://github.com/mherrmann/fbs/releases/tag/v1.1.4).
